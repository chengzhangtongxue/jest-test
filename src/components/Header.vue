<template>
    <div class="header">
        <input type="text" test-data="input" v-model="input" @keyup.enter="enterFn">
    </div>
</template>

<script>
export default {
    data() {
        return {
            input: ''
        }
    }   ,
    methods: {
        enterFn() {
            this.$emit('enterCallback');
            this.input = '';
        }
    }
}
</script>

<style lang="scss">
.header {

}
</style>