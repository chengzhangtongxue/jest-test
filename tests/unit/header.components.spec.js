import { shallowMount, mount } from '@vue/test-utils';
import Header from '@/components/Header.vue';

describe('Header.vue ', () => {
    it('header.vue 快照', () => {
        const wrapper = shallowMount(Header);
        const _input = wrapper.find('[test-data="input"]');
        expect(wrapper).toMatchSnapshot();
    })
    it('header.vue 页面有一个input 输入框', () => {
        const wrapper = shallowMount(Header);
        const _input = wrapper.find('[test-data="input"]');
        expect(_input.exists()).toBe(true);
    })

    it('header.vue 页面有一个input 默认值为空', () => {
        const wrapper = mount(Header);
        const _input = wrapper.findAll('input').at(0);
        expect(_input.element.value).toBe('');
    })

    it('header.vue data里面的input值改变，input标签值也会改变，且相等 ', () => {
        const wrapper = shallowMount(Header);
        const msg = 'msg';
        wrapper.setData({ input: msg});
        const _input = wrapper.findAll('input').at(0);
        expect(_input.element.value).toBe(msg);
    })

    it('header.vue input标签绑定了enter 时间', () => {
        const wrapper = shallowMount(Header);
        const _input = wrapper.findAll('input').at(0);
        wrapper.setData({ input: 'msg'});
        _input.trigger('keyup.enter');
        expect(_input.element.value).toBe('');
    });

})
