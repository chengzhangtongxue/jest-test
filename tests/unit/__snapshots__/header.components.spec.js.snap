// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header.vue  header.vue 快照 1`] = `<div class="header"><input type="text" test-data="input"></div>`;
